### DISCUSSION TITLE
API specs for cpp client

### DISCUSSION THREAD

**Discussion by MartyMcFlyInTheSky** [supabase/supabase#8644] [Questions]
Hi guys

I want to use supabase on an embedded system. So I write a custom c++ client. But now I need to know the API specifications for accessing supabase.

So what conversation protocol should I be following for ingoing and outgoing messages? How often the heartbeat needed etc.. Is there a spec sheet from the developers of the interface that a realtime client can follow?

**Comment by steve-chavez** [comment]
Hi there,

For storage and REST, you can follow:

- https://supabase.github.io/storage-api/
- https://postgrest.org/en/stable/api.html

Couldn't find API docs for:

- https://github.com/supabase/realtime/
- https://github.com/supabase/gotrue

You could open issues on those repos asking for the API docs.

**Comment by MartyMcFlyInTheSky** [comment]
Thanks a lot, I will check it out. Who can I get back to for minor questions?

**Reply by GaryAustin1** [reply]
You just in general ask questions here in discussions or in the Supabase discord server https://discord.com/invite/R7bSpeBSJE
Help is mainly provided by other users, or employees when they can.

Edit:  You may already know this, but looking at some of the client code here https://github.com/supabase-community and the default js repositories would also probably go a long way to understanding the api calls and code needed for things such as realtime.

**Reply by MartyMcFlyInTheSky** [reply]
Thanks, I already looked at the js and C# client for reference. Will dig into it in the next days :)

**Reply by w3b6x9** [reply]
@MartyMcFlyInTheSky and feel free to ping me here if you get stuck and I'll help out.
